src/login/logind-session-device.c

```diff
@@ -193,11 +193,19 @@ static int session_device_start(SessionDevice *sd) {
         switch (sd->type) {
 
         case DEVICE_TYPE_DRM:
-                /* Device is kept open. Simply call drmSetMaster() and hope there is no-one else. In case it fails, we
-                 * keep the device paused. Maybe at some point we have a drmStealMaster(). */
-                r = sd_drmsetmaster(sd->fd);
-                if (r < 0)
-                        return r;
+
+                if (sd->fd < 0) {
+                        /* Open device if it isn't open yet */
+                        sd->fd = session_device_open(sd, true);
+                        if (sd->fd < 0)
+                                return sd->fd;
```

Comment:
Funny, I'd say this is obviously suspicious.  I would presume it is wrong, unless I knew an explanation.  Why we do not need drmSetMaster() ?

Specifically, this leaves the master status of `fd` dependent on whether or not there was another master.  And yet we will unconditionally set `sd->active = true`, unlike the "Device is kept open" case.

Are you saying we are guaranteed not to have any master running in the scenario where we lost sd->fd ?  Why?

(And then, why is it in any way desirable to rely on this, instead of the simpler code which unconditionally calls drmSetMaster() ?).

Reply:
Doh, retracted.  s_d_o(sd, true) does do drmSetMaster().  

The comment is curious, but I'm probably better off not asking about the weird legacy DRM semantics.

                        /* Weird legacy DRM semantics might return an error even though we're master. No way to detect
                         * that so fail at all times and let caller retry in inactive state. */

Reply:
I.e. it makes me curious whether we just changed a bus call to require a retry in some situation, where it didn't before.